hooks/gen_docs/__init__.py

```diff
@@ -0,0 +1,28 @@
+"""Documentation generation."""
+
+from collections.abc import Generator
+from enum import Enum
+from typing import Any
+
+
+class SuperEnum(Enum):
+    """Adds constructors that return all items in a ``Generator``.
+
+    Introduces constructors that return a ``Generator`` object
+    either containing all items, only their names or their values.
+    """
+
+    @classmethod
+    def items(cls) -> Generator[tuple[str, Any], None, None]:
```

Comment:
if you're saying keys are always `str` shouldn't this be reflected in the class definition? otherwise I'd go for Generic